File: auto-docs/test_executables/r/axes-range-mode.r
library(plotly)

p <- plotly(username='theengineear', key='o9zlr0hy6z')

data <- list(
  list(
    x = c(2, 4, 6), 
    y = c(-3, 0, 3), 
    type = "scatter"
  )
)
layout <- list(
  xaxis = list(
    autorange = TRUE, 
    rangemode = "tozero"
  ), 
  yaxis = list(
    autorange = TRUE, 
    rangemode = "nonnegative"
  ), 
  showlegend = FALSE
)

response <- p$plotly(data, kwargs=list(layout=layout, filename="axes-range-mode", fileopt="overwrite", auto_open="FALSE"))
url <- response$url
filename <- response$filename

File: auto-docs/executables/node_js/axes_range_mode.js
// Learn about API authentication here: plot.ly/nodejs/getting-started
// Find your api_key here: plot.ly/settings/api

var plotly = require('plotly')('TestBot', 'r1neazxo9w')


var data = [
  {
    x: [2, 4, 6], 
    y: [-3, 0, 3], 
    type: "scatter"
  }
];
var layout = {
  showlegend: false, 
  xaxis: {
    rangemode: "tozero", 
    autorange: true
  }, 
  yaxis: {
    rangemode: "nonnegative", 
    autorange: true
  }
};
var graphOptions = {layout: layout, filename: "axes-range-mode", fileopt: "overwrite"};
plotly.plot(data, graphOptions, function (err, msg) {
    console.log(msg);
});
